engine/schema/src/com/cloud/upgrade/DatabaseUpgradeChecker.java

```diff
@@ -424,27 +421,9 @@ protected void upgrade(CloudStackVersion dbVersion, CloudStackVersion currentVer
                 }
 
                 upgrade.performDataMigration(conn);
-                boolean upgradeVersion = true;
-
-                if (upgrade.getUpgradedVersion().equals("2.1.8")) {
```

Comment:
I thought that anything older than 4.x should be removed. The simulator and DB schema create a structure based on 4.0.0 or 4.1.0 if I recall correctly.
I initially added another commit (removed now) which was only leaving upgrade paths from version 4.0.0. That can be done later, and I think it would be a good cleanup.